.github/workflows/publish.yml

```diff
@@ -1,54 +1,29 @@
-# Creates a git tag and publishes to pypi based on the
-# version specified in pyproject.toml.
+# publishes to pypi based on git releases. NOTE: The git tag is used to trigger the
+# workflow and should be the same as the version number in the pyproject.toml file.
 name: Publish to PyPI
 on:
   push:
-    branches:
-      - main
-    paths:
-      - pyproject.toml
-  workflow_dispatch:
-permissions:
-  contents: write
+    tags: # run on tags starting with a v character. ex. v1.0.0
+      - v*
 jobs:
   test:
     uses: ./.github/workflows/test.yml
   publish:
     needs: test
-    runs-on: ubuntu-20.04
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - name: Install poetry
         run: pipx install poetry
       - uses: actions/setup-python@v4
         with:
-          cache: 'poetry'
-          python-version: '3.11'
+          python-version: "3.x"
+          cache: "poetry"
       - name: Install dependencies
-        run: poetry install --only main
-      - name: get cvml package version
-        id: get_version
-        run: echo "::set-output name=version::$(poetry version --short)"
-      - name: check if git tag exists
-        id: check_tag
-        run: |
-          git fetch --tags
-          if git tag -l ${{ steps.get_version.outputs.version }}; then
```

Comment:
This line wasn't working. Please consider changing it to something like
```
if [[ $(git tag -l ${{ steps.get_version.outputs.version }} | wc -l) -eq '1' ]]; then
```

Reply:
Done.